### PR TITLE
DAOS-14909 pool: property for rdb svc_ops KVS entry max age

### PR DESCRIPTION
**Pull request by kccain**
To specify the maximum age of an entry in the svc_ops KVS (that is within rdb, and used for duplicate operation detection), this change introduces a pool property, DAOS_PROP_PO_SVC_OPS_ENTRY_AGE. This replaces the previous engine environment variable interface. The property may be specified during pool create and is then immutable. The default max age is 300 seconds, with a valid range of 150-600.

Features: pool

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]
Go changes LGTM

File: src/control/lib/daos/pool_property.go [hunk 1]
```diff
@@ -120,6 +120,31 @@ func PoolProperties() PoolPropertyMap {
 				valueMarshaler: numericMarshaler,
 			},
 		},
+		"svc_ops_entry_age": {
```
Comment (mjmac) [inline src/control/lib/daos/pool_property.go:123]: Would be a good idea to add some test cases for this to [TestControl_PoolProperties](https://github.com/daos-stack/daos/blob/master/src/control/lib/daos/pool_property_test.go#L79). Looks like the `svc_ops_enabled` property could use some, too.

**Review by mjmac — APPROVED** [review]
Looks good to me, just a note that it would be good to add some unit test cases for the new property. We haven't been as diligent as we should have been about insisting on these when new properties are added.

**Review by liw — APPROVED** [review]

File: src/control/lib/daos/pool_property.go (hunk 1 above)
Comment (kccain) [reply src/control/lib/daos/pool_property.go:123]: I have a separate PR with some candidate unit test cases for these two new properties, https://github.com/daos-stack/daos/pull/13676

**Pull request merged**